scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,836 @@
+            }
+            return !alreadyExistingChannel;
+        } catch (InvalidChannelException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public boolean deleteChannel(String originator, TaskId taskid, String name) throws IOException {
+        try {
+            if (persistedChannels.containsKey(name)) {
+                persistedChannels.remove(name);
+                logWithContext(taskid, null, "Deleted persistent channel " + QUOTE + name + QUOTE, Level.INFO);
+                recMan.commit();
+                return true;
+            } else if (inMemoryChannels.containsKey(name)) {
+                inMemoryChannels.remove(name);
+                logWithContext(taskid, null, "Deleted memory channel " + QUOTE + name + QUOTE, Level.INFO);
+                return true;
+            } else {
+                return false;
+            }
+        } catch (InvalidChannelException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public boolean createChannelIfAbsent(String originator, TaskId taskid, String name, boolean isPersistent)
+            throws IOException {
+        if (inMemoryChannels.containsKey(name) || persistedChannels.containsKey(name)) {
+            return false;
+        } else {
+            createChannel(originator, taskid, name, isPersistent);
+            return true;
+        }
+    }
+
+    private void commitIfNeeded(String channel) throws IOException {
+        if (persistedChannels.containsKey(channel)) {
+            persistedChannels.put(channel, persistedChannels.get(channel));
+            recMan.commit();
+        }
+    }
+
+    @Override
+    public boolean containsKey(String originator, TaskId taskid, String channel, String key)
+            throws InvalidChannelException {
+        return getChannel(channel).containsKey(key);
+    }
+
+    @Override
+    public int size(String originator, TaskId taskid, String channel) throws InvalidChannelException {
+        return getChannel(channel).size();
+    }
+
+    @Override
+    public boolean isEmpty(String originator, TaskId taskid, String channel) throws InvalidChannelException {
+        return getChannel(channel).isEmpty();
+    }
+
+    @Override
+    public boolean containsValue(String originator, TaskId taskid, String channel, Serializable value)
+            throws InvalidChannelException {
+        return getChannel(channel).containsValue(value);
+    }
+
+    @Override
+    public <T extends Serializable> T get(String originator, TaskId taskid, String channel, String key)
+            throws InvalidChannelException {
+        return ((Channel<T>) getChannel(channel)).get(key);
+    }
+
+    @Override
+    public <T extends Serializable> T put(String originator, TaskId taskid, String channel, String key, T value)
+            throws InvalidChannelException, IOException {
+        T answer = ((Channel<T>) getChannel(channel)).put(key, value);
```

Comment:
What happens when `getChannel(channel)` is not `Channel<T>`?

Reply:
ClassCastException ?

Reply:
Sorry, my mistake. I wanted to removed my comment before you answer it)